pkg/vsphere/disk/vmdk.go

```diff
@@ -66,8 +66,9 @@ func LockedVMDKFilter(vm *mo.VirtualMachine) bool {
 }
 
 func IsLockedError(err error) bool {
-	// TODO: add check here for actual error
-	return true
+	disks := LockedDisks(err)
+	//if device is locked, disks will not be nil
+	return disks != nil
```

Comment:
this is correct, but a more implementation tolerant check is below - this will handle both `nil` and the case of an empty array:
```go
len(disks) > 0
```